dsn_test.go

```diff
@@ -1639,3 +1639,19 @@ func TestExtractAccountName(t *testing.T) {
 		})
 	}
 }
+
+func TestDecodeDSNParam(t *testing.T) {
+	testcases := map[string]string{
+		"query_tag":             "query_tag",
+		"%24my_custom_variable": "$my_custom_variable",
+	}
+
+	for param, expected := range testcases {
+		t.Run(param, func(t *testing.T) {
+			decodedParam := urlDecodeIfNeeded(param)
+			if decodedParam != expected {
+				t.Fatalf("urlDecodeIfNeeded returned unexpected response (%v), should be %v", decodedParam, expected)
```

Comment:
Can we use `assertNilE` here?

Reply:
i'm probably missing something here :( `urlDecodeIfNeeded` doesn't return `nil` so not sure why we want to check that?

Reply:
Sorry, I meant `assertEqualE` :)

Reply:
oh, got it, replaced